docs/deployment/deploy_gardenlet.md

```diff
@@ -0,0 +1,344 @@
+# Deploy a Gardenlet
+
+This document describes how to manually deploy a Gardenlet into a Kubernetes cluster and how to automatically register the cluster as a Seed.
+
+**When is this useful**
+
+- When deploying the first Gardenlet into a Gardener installation (this is done automatically by [garden-setup](https://github.com/gardener/garden-setup))
+- To register a Kubernetes cluster as a `Seed` cluster that is in a "fenced" environment, such as a private datacenter or behind a firewall.
+
+Elaborating on the latter, the Gardenlet only requires network connectivity from the Gardenlet to the Garden cluster (not the other way round), so it can be used to register Kubernetes clusters with no public endpoint. 
+When registering a Seed cluster in such a "fenced" environment, the [automatic Gardenlet deployment into Shooted Seed clusters](https://github.com/gardener/gardener/blob/master/docs/usage/shooted_seed.md) cannot be used (see [additional consideration](#additional-considerations)).
+
+Additionally, the Gardenlet can automatically register a Kubernetes cluster as a Seed cluster with the Gardener based on a pre-defined Seed specification (see [automatically register as a Seed cluster](#automatically-register-as-a-seed-cluster)).
+
+## Additonal considerations
+
+The Gardenlet can **automatically** deploy itself into `Shoot` clusters and register this cluster as a `Seed` (these clusters are called `Shooted Seeds` in the Gardener terminology). This is the preferred way to add additional Seed clusters.
+
+This works by registering an initial Cluster as a Seed cluster (you can use the Garden cluster itself!) - we call it the `Soil` - that has a Gardenlet already installed (e.g by following this guide). The `Soil` cluster is then used as the `Seed` cluster for the `Shooted Seed` clusters.  
```

Comment:
The term "soil" is never used anywhere yet, so I'd rather try to not introduce it as it increases the complexity and makes it harder to understand. Just use garden/seed/shoot. You can use "initial cluster" instead, if you like.

Reply:
Removed mentions of "soil"